content/blog/2022-03-31-stale-models.md

```diff
@@ -0,0 +1,223 @@
+┡━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━━━╇━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━┩
+│ workspace               │  0.70164 │ 0.51384 │ 20210428   │ 450         │ 64              │
+│ main                    │  0.60791 │ 0.45758 │ 20210428   │ 375         │ 64              │
+│ └── 801fdff [exp-a80c0] │  0.70164 │ 0.51384 │ 20210428   │ 450         │ 64              │
+└─────────────────────────┴──────────┴─────────┴────────────┴─────────────┴─────────────────┘
+```
+
+This model doesn't have the best metrics, so we can run more experiments to see
+if tuning hyperparameters will help before we deploy this model to production.
+Let's change the values of the `train.n_est` and `train.n_est` hyperparameters.
+We'll run several experiments with different values and it will produce a table
+similar to this:
+
+```dvctable
+┏━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━┓
+┃ neutral:**Experiment**              ┃ metric:**avg_prec** ┃ metric:**roc_auc** ┃ param:**train.seed** ┃ param:**train.n_est** ┃ param:**train.min_split** ┃
+┡━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━━━╇━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━┩
+│ workspace               │  0.43501 │ 0.79082 │ 20210428   │ 475         │ 28              │
+│ main                    │  0.60791 │ 0.45758 │ 20210428   │ 375         │ 64              │
+│ ├── 78d29aa [exp-f06bb] │  0.43501 │ 0.79082 │ 20210428   │ 475         │ 28              │
+│ ├── 8fb41cf [exp-1323d] │  0.42796 │ 0.80841 │ 20210428   │ 425         │ 28              │
+│ ├── 434a82f [exp-63459] │  0.36044 │ 0.87037 │ 20210428   │ 350         │ 28              │
+│ ├── 549586e [exp-ceb6d] │  0.61998 │  0.4306 │ 20210428   │ 350         │ 64              │
+│ ├── fbf8760 [exp-affe2] │  0.68824 │ 0.50067 │ 20210428   │ 425         │ 64              │
+│ ├── 732ab92 [exp-f8e8d] │  0.65138 │ 0.49431 │ 20210428   │ 500         │ 64              │
+│ └── 801fdff [exp-a80c0] │  0.70164 │ 0.51384 │ 20210428   │ 450         │ 64              │
+└─────────────────────────┴──────────┴─────────┴────────────┴─────────────┴─────────────────┘
+```
+
+We've run a few experiments with a different hyperparameter value each time and
+it looks like `exp-63459` is the best one out of them based on both average
+precision and the ROC-AUC value. So we'll apply this experiment to our workspace
+and choose this model as the one that will go to production. To apply the
+experiment, we'll run the following command:
+
+```dvc
+$ dvc exp apply exp-c85c3
+```
+
+This will update the workspace with the exact code, data, and hyperparameters
+that were used in that particular experiment. So we can commit these changes to
+Git and we'll have a reference to everything we need for this exact model. Now
+let's say we have deployed this to production and it's been a great model for
+almost another year, then we start noticing data drift again.
```

Comment:
Why "again"? It's not very clear from this section why we're taking these steps. Is it to simulate an "old" model and then see how it will drift later with the 2012 data (next section) ? Also why 2011 and 2012? 🙂

Reply:
It's a part of the overall story we're telling about data drift. Earlier in the post, we reference two different dataset updates so that's why we're saying "again". And 2011 and 2012 are just the years for the dataset we could use.